docs/QUICKSTART.md

```diff
@@ -0,0 +1,418 @@
+
+Inside `my-fuel-dapp/frontend` run;
+
+```sh
+npx typechain --target=fuels --out-dir=./src/contracts ../contract/out/debug/*-abi.json
+```
+
+You should see something like this;
+
+```sh
+Successfully generated 4 typings!
+```
+
+Now you should be able to find a new folder `my-fuel-dapp/frontend/src/contracts`. This folder was auto-generated by our `typechain` command, this files abstract the work we would need to do, to create a contract instance, and also generate a complete TypeScript interface to the Contract, making easy to develop.
+
+### 3. Modify the App.
+
+Inside the `frontend` folder let's add a code that interact with our contract.
+Read the comments to help you understand the App parts.
+
+Change the file `my-fuel-dapp/frontend/src/App.tsx` to;
+
+```ts
+import React, { useEffect, useState } from "react";
+import { Wallet } from "fuels";
+import "./App.css";
+// Import the contract factory from the generated folder
+// from the typechain command
+import { ContractsAbi__factory } from "./contracts";
+
+// Se the Wallet Secret used on the chainConfig.json
+// this enables us to have a account with initial balance
+const WALLET_SECRET =
+  "0xa449b1ffee0e2205fa924c6740cc48b3b473aa28587df6dab12abc245d1f5298";
+// The address of the contract deployed to our local node
+// the contract id is output right after the forc deploy command
+const CONTRACT_ID =
+  "0xa326e3472fd4abc417ba43e369f59ea44f8325d42ba6cf71ec4b58123fd8668a";
+// Create a Wallet from given secretKey in this case
+// The one we configured at the chainConfig.json
+const wallet = new Wallet(WALLET_SECRET);
+// Connects out Contract instance to the deployed contract
+// address using the given wallet.
+const contract = ContractsAbi__factory.connect(CONTRACT_ID, wallet);
+
+function App() {
+  const [counter, setCounter] = useState(0);
+
+  useEffect(() => {
+    async function main() {
+      // Executes the counter function to query the current contract state
+      // the `.get()` is read-only, because of this it don't expand coins.
+      const { value } = await contract.functions.counter().get();
+      setCounter(Number(value));
+    }
+    main();
+  }, []);
+
+  async function increment() {
+    // Creates a transactions to call the increment function passing the amount
+    // we want to increment, because it creates a TX and updates the contract state
+    // this requires the wallet to have enough coins to cover the costs and also
+    // to sign the Transaction
+    const { value } = await contract.functions.increment(1).call();
+    setCounter(Number(value));
+  }
+
+  return (
+    <div className="App">
+      <header className="App-header">
+        <p>Counter: {counter}</p>
+        <button onClick={increment}>Increment</button>
+      </header>
+    </div>
+  );
+}
+
+export default App;
+```
+
+### 4. Run your project.
+
+Now it's time to have fun run the project on your browser;
+
+Inside `my-fuel-dapp/frontend` run;
+
+```sh
+npm start
```

Comment:
When running this command I get the following error: 
```
Compiled with problems:
ERROR in ./src/App.js 11:0-52

Module not found: Error: Can't resolve './contracts' in '/Users/mattauer/src/my-fuel-dapp/frontend/src'
```
I think this is probably because I have an App.js file locally instead of App.tsx and a tsconfig.  Is their a create-react-app command that generate a typescript project?

Reply:
@matt-user thanks I forgot to add the `--template` on the command;

Correct one:
```sh
npx create-react-app frontend --template typescript
```